index.js

```diff
@@ -0,0 +1,97 @@
+const { join } = require('path');
+const moduleAlias = require('module-alias');
+
+function npm(name) {
+  return join(process.cwd(), 'node_modules', name);
+}
+
+moduleAlias.addAliases({
+  react: npm('preact/compat'),
+  'react-dom': npm('preact/compat'),
+  'react-ssr-prepass': npm('preact-ssr-prepass'),
+  webpack: npm('webpack')
+});
+
+// this has to come after the webpack alias is set up:
+const withPrefresh = require('@prefresh/next');
+
+for (const dep of ['preact', 'preact-render-to-string']) {
+  try {
+    require.resolve(dep);
+  } catch (e) {
+    throw Error(`[preact] Missing "${dep}" dependency:\n  npm i ${dep}`);
+  }
+}
+
+const pkg = require(join(process.cwd(), 'package.json'));
+const deps = pkg.dependencies;
+if (deps) {
+  let toInstall = [];
+  if (!deps.react || /^[\^~\d]/.test(deps.react)) {
+    toInstall.push('react@npm:@preact/compat');
+  }
+  if (!deps['react-dom'] || /^[\^~\d]/.test(deps['react-dom'])) {
+    toInstall.push('react-dom@npm:@preact/compat');
+  }
+  if (toInstall.length) {
+    throw (
+      `[preact] Missing/incorrect alias dependencies.\nPlease run:\n` +
+      `  npm i ${toInstall.join(' ')}`
+    );
```

Comment:
```suggestion
    console.error(
      `[preact] Missing/incorrect alias dependencies.\nPlease run:\n` +
      `  npm i ${toInstall.join(' ')}`
    );
    process.exit(-1);
```

to suppress a stacktrace